docs/developers/library_examples.md

```diff
@@ -0,0 +1,960 @@
+# Library Examples
+
+## C API
+
+### Connecting to an iRODS server
+
+Demonstrates how to use `rcConnect`, `clientLogin`, and `rcDisconnect`.
+
+```c++
+#include <irods/getRodsEnv.h>
+#include <irods/rodsClient.h>
+#include <irods/rcConnect.h>
+
+void connecting_to_an_irods_server()
+{
+    rodsEnv env;
+
+    if (const int ec = getRodsEnv(&env); ec != 0) {
+        // Failed to load the local environment information (i.e. irods_environment.json).
+        // Handle error.
+        return;
+    }
+
+    // This is the safest place to load all client-side API plugins.
+    // This step is required for proper interaction with a 4.3.0 server. Forgetting to call this
+    // function will result in errors if subsequent API calls depend on client-side plugins.
+    load_client_api_plugins();
+
+    rErrMsg_t error;
+    RcComm* conn = rcConnect(env.rodsHost,
+                             env.rodsPort,
+                             env.rodsUserName,
+                             env.rodsZone,
+                             0,
+                             &error);
+
+    if (!conn) {
+        // Failed to connect to server.
+        // Handle error.
+        return;
+    }
+
+    // We've successfully connected to an iRODS server.
+    // Here's how you authenticate (i.e. log in) with the server.
+    if (const int ec = clientLogin(conn); ec != 0) {
+        // Failed to authenticate with server.
+        // Handle error and disconnect.
+        rcDisconnect(conn);
+        return;
+    }
+
+    //
+    // Use the "conn" pointer to execute operations.
+    //
+
+    // When done, call rcDisconnect() so that the server knows the client is finished.
+    // This lets the server know it is okay to deallocate resources.
+    rcDisconnect(conn);
+}
+```
+
+### Iterating over a collection (client-side)
+
+Demonstrates how to use `rclOpenCollection`, `rclReadCollection` and `rclCloseCollection`.
+
+```cpp
+#include <irods/miscUtil.h>
+
+#include <iostream>
+
+void iterating_over_a_collection()
+{
+    RcComm* conn = // Our iRODS connection.
```

Comment:
one of 8 in this page - no right hand side?

Reply:
Done.